cpp/CMakePresets.json

```diff
@@ -112,8 +112,8 @@
       "generator": "Unix Makefiles",
       "cacheVariables": {
         "CMAKE_MAKE_PROGRAM": "make",
-        "CMAKE_C_COMPILER": "clang",
-        "CMAKE_CXX_COMPILER": "clang",
+        "CMAKE_C_COMPILER": "$env{CC}",
+        "CMAKE_CXX_COMPILER": "$env{CXX}",
```

Comment:
These are the [default](https://cmake.org/cmake/help/latest/envvar/CC.html), so you can simply delete these lines.